src/plugins/foxPage/hooks/useFarmingApr.ts

```diff
@@ -17,13 +24,55 @@ import { useCurrentBlockNumber } from './useCurrentBlockNumber'
 
 const ethersProvider = getEthersProvider()
 
+function calculateSlippageMargin(percentage: number, amount: string) {
+  const remainingPercentage = (100 - percentage) / 100
+  return bnOrZero(toBaseUnit(amount, 18))
+    .times(bnOrZero(remainingPercentage))
+    .decimalPlaces(0)
+    .toFixed()
+}
+
+function bufferGas(limitOrPrice: string) {
+  return bnOrZero(limitOrPrice).times(1.2).decimalPlaces(0).toFixed()
+}
+
+async function getBufferedGas(tx: TransactionRequest) {
+  let gasLimit = null
+  let gasPrice = null
+  try {
+    if (ethersProvider && tx) {
+      const gas = await ethersProvider?.estimateGas(tx)
+      const price = await ethersProvider?.getGasPrice()
+      if (gas && price) {
+        gasLimit = bufferGas(gas?.toString())
+        gasPrice = bufferGas(price?.toString())
+      }
+    }
+    return {
+      gasLimit,
+      gasPrice,
+    }
+  } catch {
+    throw Error('Problem estimating gas')
+  }
+}
+
+const toHex = (value: BigNumberish) => BigNumber.from(value).toHexString()
+
+const slippagePercentage = 3
+
 export const useFarmingApr = () => {
   const [farmingAprV2, setFarmingAprV2] = useState<string | null>(null)
   const [farmingAprV4, setfarmingAprV4] = useState<string | null>(null)
   const [isFarmingAprV2Loaded, setIsFarmingAprV2Loaded] = useState(false)
   const [isFarmingAprV4Loaded, setIsFarmingAprV4Loaded] = useState(false)
   const blockNumber = useCurrentBlockNumber()
 
+  const accountIds = useAppSelector(state =>
+    selectAccountIdsByAssetId(state, { assetId: 'eip155:1/slip44:60' }),
+  )
+  const connectWalletEthAddress = accountIds[0]
+
```

Comment:
Use `selectFirstAccountSpecifierByChainId` which will give you an actual account, not a CAIP-vernacular `AccountId`, which is what you ned here